src/core/qgsfield.h

```diff
@@ -215,6 +216,42 @@ class CORE_EXPORT QgsField
      */
     QString comment() const;
 
+    /**
+     * Returns the map of field metadata.
+     *
+     * Map keys should match values from the Qgis::FieldMetadataProperty enum.
+     *
+     * \see setMetadata()
+     * \since QGIS 3.30
+     */
+    QMap< int, QVariant > metadata() const;
+
+    /**
+     * Returns a specific metadata \a property.
+     *
+     * \see setMetadata()
+     * \since QGIS 3.30
+     */
+    QVariant metadata( Qgis::FieldMetadataProperty property ) const;
```

Comment:
will users be really able to use this method (as well as the corresponding setter), for a CustomProperty + 1 value that will not be in the FieldMetadataProperty enum class ?

Reply:
I was also wondering about this. An example of that in the added test would help.

By the way, the currently proposed syntax looks convoluted to me, but maybe the idea is to have wrappers for the various properties ?

> field.setMetadata( {{ static_cast< int >( Qgis::FieldMetadataProperty::GeometryCrs ), QStringLiteral( "abc" )}, {2, 5 }} );
> field.setMetadata( Qgis::FieldMetadataProperty::GeometryWkbType, QStringLiteral( "def" ) );

Reply:
Fixed now, with added tests